### PR TITLE
Update/audit external links 

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Some easy wins in updating external links for September docs housekeeping.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/console-deployment/overview.md [hunk 1]
```diff
@@ -8,5 +8,3 @@ description: "Console Self-Managed provides key insights into automation cluster
 Camunda Console Self-Managed offers a centralized overview for Camunda 8 installations, designed to enhance operational efficiency and oversight within Enterprise environments. Console Self-Managed provides key insights into automation cluster deployments, process orchestration usage, and streamlining usage tracking.
 
 Camunda Console Self-Managed provided is available as a container image. Refer to the [installation guide](/self-managed/setup/overview.md) for details on how to install this component.
-
-Participate in development by sharing your feedback on our [ProductBoard page](https://portal.productboard.com/q4b4cm3z3fkx5zesq51o3mz3).
```
Comment (akeller) [inline docs/self-managed/console-deployment/overview.md:12]: @javi-alzorriz-camunda do we want to provide an opportunity for people to provide feedback via ProductBoard?

**Review by akeller — APPROVED** [review]
I left a non-blocking comment, but I'm fine with this merging.

File: docs/self-managed/console-deployment/overview.md (hunk 1 above)
Comment (conceptualshark) [reply docs/self-managed/console-deployment/overview.md:12]: If we need to swap something new in here, I can follow-up with a separate PR to update it. 👍

Comment (akeller) [reply docs/self-managed/console-deployment/overview.md:12]: @conceptualshark, We should not use ProductBoard in the public-facing docs; it's for internal use only. Instead, non-Camundi would follow the feature request process in JIRA. I don't think we have any more links in ProductBoard and this is covered on the [contact page](https://docs.camunda.io/contact/).

**Pull request merged**